packages/UI/src/components/ShareDialog/ShareDialog.tsx

```diff
@@ -0,0 +1,169 @@
+import {
+  Twitter,
+  Facebook,
+  Instagram,
+  Linkedin,
+  Link as LinkIcon,
+} from 'lucide-react';
+import {
+  FacebookShareButton,
+  TwitterShareButton,
+  LinkedinShareButton,
+  InstapaperShareButton,
+} from 'next-share';
+import { tv } from 'tailwind-variants';
+import Button from '../atoms/Button';
+import {
+  Dialog,
+  DialogTrigger,
+  DialogContent,
+  DialogHeader,
+  DialogFooter,
+  DialogTitle,
+} from './../Dialog';
+
+const ShareDialogStyles = tv(
+  {
+    slots: {
+      dialogHeader:
+        'flex h-6 w-[360px] items-center justify-center font-nunito-sans text-xl uppercase tracking-wider',
+      dialogFooter:
+        'flex w-[320px] flex-col items-center justify-end gap-4 font-nunito-sans',
+      dialogBody: 'flex w-[320px] items-center justify-evenly gap-4',
+      shareBtn: 'flex h-10 w-10 items-center justify-center rounded shadow-lg',
+    },
+    variants: {
+      screenPseudo: {
+        md: {
+          dialogHeader: 'w-[448px]',
+        },
+        lg: {},
+        xl: {},
+      },
+    },
+  },
+  {
+    responsiveVariants: ['md', 'lg', 'xl'],
+  }
+);
+
+export interface ShareDialogProps {
+  /*
+   * The header of the dialog. For example, "Idea" or "Project".
+   */
+  header: string;
+  /*
+   * The title of the shared content. Defaults to the current page title.
+   */
+  title?: string;
+  /*
+   * The description of the shared content. Defaults to the current page description.
+   */
+  description?: string;
+  /*
+   * The quote of the shared content. Defaults to the current page description.
+   */
+  quote?: string;
+  /*
+   * The hashtag to share. Defaults to the current page hashtag.
+   */
+  hashtag?: string;

```

Comment:
how does this work? does it add a hashtag from the current url?

`https://google.com/something#segment`
‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎       ⤡ is this the hashtag in question?

Reply:
Facebook is the only one that uses the hashtag.
https://www.facebook.com/formedia/blog/using-hashtags-on-facebook

Reply:
Oh, I guess I was looking at my old code and thought it was required. 

Per documentation hashtag prop is optional for both social media. Facebook's hashtag prop is a string, while twitter's is an array. 

I'll add this functionality